valohai/internals/download.py

```diff
@@ -14,21 +45,34 @@ def download_url(url: str, path: str, force_download: bool = False) -> str:
                 f"for download support (attempting to download {url})"
             ) from ie
 
-        tmp_path = tempfile.NamedTemporaryFile().name
-        print(f"Downloading {url} -> {path}")  # noqa
-        r = requests.get(url, stream=True)
-        r.raise_for_status()
-        total = (
-            int(r.headers["content-length"]) if "content-length" in r.headers else None
-        )
-
-        with tqdm(total=total, unit="iB", unit_scale=True) as t, open(
-            tmp_path, "wb"
-        ) as f:
-            for data in r.iter_content(1048576):
-                t.update(len(data))
-                f.write(data)
-        os.replace(tmp_path, path)
+        if "datum://" in url:
+            input_folder_path = path.rsplit("/", 1)[0]  # getting folder path only
+            response = resolve_datum(uri_to_filename(url))
+            if response.status_code == 200:
```

Comment:
This is a bit of an useless check, since you're calling `raise_for_status` in `resolve_datum` (so error codes would never get here).